frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -270,46 +273,73 @@ public void queueRepositoryUpdateAndModelRefresh(@UI Runnable doOnUIThreadWhenRe
         Case($(), (Object) null));
   }
 
-  private static class GitMacheteGraphTableMouseAdapter extends MouseAdapter {
-    private final EnhancedGraphTable outer;
+  private static class EnhancedGraphTableMouseAdapter extends MouseAdapter {
+    private final EnhancedGraphTable graphTable;
+    private final EnhancedGraphTablePopupMenuListener popupMenuListener;
 
     @UIEffect
-    GitMacheteGraphTableMouseAdapter(EnhancedGraphTable outer) {
-      this.outer = outer;
+    EnhancedGraphTableMouseAdapter(EnhancedGraphTable graphTable) {
+      this.graphTable = graphTable;
+      this.popupMenuListener = new EnhancedGraphTablePopupMenuListener(graphTable);
     }
 
     @Override
     @UIEffect
     public void mouseClicked(MouseEvent e) {
       Point point = e.getPoint();
-      int row = outer.rowAtPoint(point);
-      int col = outer.columnAtPoint(point);
+      int row = graphTable.rowAtPoint(point);
+      int col = graphTable.columnAtPoint(point);
 
       // check if we click on one of branches
       if (row < 0 || col < 0) {
         return;
       }
 
-      BranchOrCommitCell cell = (BranchOrCommitCell) outer.getModel().getValueAt(row, col);
+      BranchOrCommitCell cell = (BranchOrCommitCell) graphTable.getModel().getValueAt(row, col);
       IGraphItem graphItem = cell.getGraphItem();
       if (!graphItem.isBranchItem()) {
         return;
       }
 
-      outer.selectedBranchName = graphItem.asBranchItem().getBranch().getName();
+      graphTable.selectedBranchName = graphItem.asBranchItem().getBranch().getName();
 
       ActionManager actionManager = ActionManager.getInstance();
       if (SwingUtilities.isRightMouseButton(e)) {
         ActionGroup contextMenuActionGroup = (ActionGroup) actionManager.getAction(ActionGroupIds.ACTION_GROUP_CONTEXT_MENU);
         var actionPopupMenu = actionManager.createActionPopupMenu(ACTION_PLACE_CONTEXT_MENU, contextMenuActionGroup);
-        actionPopupMenu.getComponent().show(outer, (int) point.getX(), (int) point.getY());
+        JPopupMenu popupMenu = actionPopupMenu.getComponent();
+        popupMenu.addPopupMenuListener(popupMenuListener);
+        popupMenu.show(graphTable, (int) point.getX(), (int) point.getY());
       } else if (SwingUtilities.isLeftMouseButton(e) && e.getClickCount() == 2 && !e.isConsumed()) {
         e.consume();
-        DataContext dataContext = DataManager.getInstance().getDataContext(outer);
+        DataContext dataContext = DataManager.getInstance().getDataContext(graphTable);
         var actionEvent = AnActionEvent.createFromDataContext(ACTION_PLACE_CONTEXT_MENU, new Presentation(), dataContext);
         actionManager.getAction(ACTION_CHECK_OUT).actionPerformed(actionEvent);
       }
     }
   }
 
+  private static class EnhancedGraphTablePopupMenuListener implements PopupMenuListener {
```

Comment:
Not a major problem... but I can see that the row is first becoming dark blue for a fraction of the second and then turning gray... do you see it this way as well? is it intended?

Reply:
Yes, I see. No, this is not intended. It's probably because of a focus transfer from the row to the context menu: row selection appear before a context menu is created and showed. After context menu is showed then focus is transferred to it and row selection became darker.